### ISSUE TITLE
IndexedDB caching of Meteor 1.5 dynamic modules suddenly failing in Chrome 59

### ISSUE THREAD

**Issue by benjamn** [meteor/meteor#8685]
Just after publishing [the first RC of Meteor 1.5](https://github.com/meteor/meteor/pull/8327#issuecomment-300588026) in #8327, I noticed a fairly significant regression, which is that the IndexedDB cache for dynamic modules is failing to populate in Chrome (version 59). Everything about [this `cache.setMany` call](https://github.com/meteor/meteor/blob/74c64369d426998706d923e7f08791b43ede1231/packages/dynamic-import/client.js#L62) appears to succeed, but nothing ends up in the database.

This problem does not seem to affect other browsers (I tried Firefox and Safari), and it does not appear to be related to the Meteor 1.5 beta/RC version, since the same problem now occurs in Chrome for the [app](http://meteor-night-dynamic-imports-demo.meteorapp.com/) that I deployed for my Meteor Night demo more than two months ago.

Can anyone else reproduce this problem? Remember that you'll need to `meteor run --production` to enable caching, and if you've been testing Meteor 1.5 already, you should probably delete any existing `MeteorDynamicImportCache` database (don't worry, it's harmless) to avoid using cache entries that were written before this problem appeared.

**Comment by benjamn** [comment]
The exact Chrome version was 59.0.3071.15, though I just tried updating to 60.0.3088.3 and the problem is still happening.

**Comment by hwillson** [comment]
Everything appears to be working properly for me with Chrome 58. I did have problems with Chrome 57, but after updating everything looks good. Fingers crossed this is just a beta/dev channel issue with versions 59/60. Chrome added IndexedDB 2.0 support in 58, so hopefully they've just broken something in the dev versions, that will be fixed.

**Comment by abernix** [comment]
I'm on Chrome 60.0.3095.0 (Canary).

I cleared my IndexedDB for `MeteorDynamicImportCache` (which previously had content in the `sourcesByVersion` object store) and visited http://meteor-night-dynamic-imports-demo.meteorapp.com/.  Dynamic imports worked properly, and upon hitting the "Refresh" button on the `sourcesByVersion` object store, it was repopulated with data.

**Comment by klaussner** [comment]
I just tried it with the same Chrome version as @abernix (and a new Meteor app using the release candidate) and it worked as expected.

**Comment by benjamn** [comment]
I think I've figured out the problem!

This morning I was excited to see that MS Visual Studio had been released for the Macintosh: https://www.visualstudio.com/vs/visual-studio-mac/

I began downloading it, not fully realizing how large it was, and lo! I began to receive warnings of "low disk space" and "insufficient swap space" from my operating system. I canceled the installation, but was warned that installation progress would be saved on my machine, in case I decided to resume installing MSVS at a later time.

I should have heeded that warning, because apparently Chrome disables/evicts IndexedDB databases when the host machine is running low on disk space! I'd seen [this page](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Browser_storage_limits_and_eviction_criteria) about quotas before, but I never imagined it would ever affect *me*.

Incredibly, after I deleted a few large files from my hard drive, dynamic module caching in Meteor 1.5 began working again. Just for fun, I was also able to reproduce the problem by filling my hard drive with empty files:
```sh
cd /tmp
mkfile -n 15g temp_15GB_file
```

This internal URL gives some information about these limits: chrome://quota-internals

The good news is that nothing was ever broken in Meteor; the app just had to re-download files it should have found in the IndexedDB cache.

**Comment by hwillson** [comment]
> Incredibly, after I deleted a few large files from my hard drive

So you deleted your `.meteor/local/bundler-cache` directory? 😄 
(Gotta love Meteor humor - I'm here all week!)

**Comment by vladejs** [comment]
@benjamn , I'm getting the following error on Firefox v48 when using Private Browsing. My app just *can't work*:
```
InvalidStateError102c5ea49c814b454f5b4e1af725cbc3fb9827d9.js:87:2256
Error: IndexedDB failure in indexedDB.open {}
```
I think Firefox disables IndexedDB on that mode, but I am not sure.
It doesn't happen on Chrome Incognito though.

**Comment by benjamn** [comment]
@vladejs Thanks for reporting that problem! I've opened a new issue to make sure it gets fixed before 1.5 is released: https://github.com/meteor/meteor/issues/8697
